src/sagas/workspaces.ts

```diff
@@ -279,7 +282,7 @@ export default function* workspaceSaga(): SagaIterator {
     const globals: Array<[string, any]> = yield select(
       (state: IState) => (state.workspaces[workspaceLocation] as IWorkspaceState).globals
     );
-    if (newChapter !== oldChapter) {
+    if (newChapter !== oldChapter || newVariant !== oldVariant) {
```

Comment:
Reset the context upon a variant change as well, in addition to a chapter change (pre-existing)